### PR TITLE
fix(backend): 🍰 Fix unintentional exposure of reports

### PR DESCRIPTION
**Pull request by Tirokk**
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-17T11:41:39Z" title="Monday, February 17th 2020, 12:41:39 pm +01:00">Feb 17, 2020</time>_
_Closed <time datetime="2020-02-21T11:07:19Z" title="Friday, February 21st 2020, 12:07:19 pm +01:00">Feb 21, 2020</time>_
---

## 🍰 Pullrequest

Hot fix: Set sensible report properties to moderators only in permissionsMiddleware.

### Issues

- relates #3074
- Original PR #3075

### Todo

None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-02-17T12:28:48Z" title="Monday, February 17th 2020, 1:28:48 pm +01:00">Feb 17, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/5312/specs"></a> <b title="Passed">61</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/5312/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 21fd3e077c |
| **Started** | Feb 19, 2020 8:45 AM |
| **Ended** | Feb 19, 2020 8:59 AM |
| **Duration** | 14:16 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Electron 78  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/5312/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-17T19:47:29Z" title="Monday, February 17th 2020, 8:47:29 pm +01:00">Feb 17, 2020</time>_

---


hey @Tirokk ... I tried to add back the tests, but I also refactored to return the `resource` instead of needlessly exposing the `Report`, without just cause.
I hope I am not stepping on your toes. I know that you said you had another `PR` that you are working on, or one in mind... we have been dealing with `build` issues since last `Friday`, at least, so we couldn't deploy any fix, until we get it figured out. 
I guess we can talk about this approach and see if people like it, or have other ideas and go from there. 
Again, sorry if I haven't understood what you meant, maybe you could let me know if there is any way I can improve this. I have reread your comments in `Discord` and I am still lost, to be honest.

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-18T15:25:20Z" title="Tuesday, February 18th 2020, 4:25:20 pm +01:00">Feb 18, 2020</time>_

---


please let's discuss this in the daily tomorrow?

File: backend/src/schema/resolvers/reports.js [hunk 1]
```diff
@@ -38,12 +29,11 @@ export default {
...
-        return reportTransactionResponse.records.map(transformReturnType)
+        return reportTransactionResponse.records.map(record => record.get('returnedResource'))
       })
       try {
-        const [createdRelationshipWithNestedAttributes] = await reportWriteTxResultPromise
-        if (!createdRelationshipWithNestedAttributes) return null
-        return createdRelationshipWithNestedAttributes
+        const [reportedResource] = await reportWriteTxResultPromise
+        return reportedResource || null
       } finally {
```
Comment (Tirokk) [inline backend/src/schema/resolvers/reports.js:37]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-18T16:07:13Z" title="Tuesday, February 18th 2020, 5:07:13 pm +01:00">Feb 18, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -38,12 +29,12 @@ export default {
           },
         )
         log(reportTransactionResponse)
-        return reportTransactionResponse.records.map(transformReturnType)
+        return reportTransactionResponse.records.map(record => record.get('returnedResource'))
       })
       try {
-        const [createdRelationshipWithNestedAttributes] = await reportWriteTxResultPromise
-        if (!createdRelationshipWithNestedAttributes) return null
-        return createdRelationshipWithNestedAttributes
+        const [reportedResource] = await reportWriteTxResultPromise
+        if (!reportedResource) return null
+        return reportedResource
```


```suggestion
        return reportedResource || null
```
If you want to convert `undefined` to `null`.

Comment (Tirokk) [reply backend/src/schema/resolvers/reports.js:37]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-18T20:53:25Z" title="Tuesday, February 18th 2020, 9:53:25 pm +01:00">Feb 18, 2020</time>_

---


that makes sense!

Comment (Tirokk) [reply backend/src/schema/resolvers/reports.js:37]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-19T07:40:07Z" title="Wednesday, February 19th 2020, 8:40:07 am +01:00">Feb 19, 2020</time>_

---


Yes, I like this better! 😍 @roschaefer

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-21T11:07:19Z" title="Friday, February 21st 2020, 12:07:19 pm +01:00">Feb 21, 2020</time>_

---


Will be solved as a whole in #3075

**Review by Mogge — APPROVED** [review]
Hm, Ok, fair enough. We could also (for the time being) make the user who reports a resource a moderator.

**Pull request closed without merging**